controller/information_source/manager.py

```diff
@@ -120,9 +123,24 @@ def update_information_source(
 
 
 def delete_information_source(project_id: str, source_id: str) -> None:
+    if information_source.get(
+        project_id, source_id
+    ).type == enums.InformationSourceType.ACTIVE_LEARNING.value and config_service.get_config_value(
```

Comment:
to prevent edge cases i would get the is from the database first and only continue if != None

```suggestion
    is_item = information_source.get(project_id, source_id)
    if not is_item:
         //some error
         return
    if is_item.type == enums.InformationSourceType.ACTIVE_LEARNING.value and config_service.get_config_value(
```

Reply:
changed to the proposed solution